javascript/index.js

```diff
@@ -4,37 +4,163 @@
+                  
+            });
+          });
+        });
+      });
+    });
+  });
+});
+
+// Iteration 2 - using promises
+obtainInstruction('steak', 0)
+  .then( (step0) => {
+    document.querySelector("#steak").innerHTML += `<li>${step0}</li>`
+    return 1
+}).then ((step1) => {
+    return obtainInstruction('steak', step1)
+      .then((step) => {
+        document.querySelector("#steak").innerHTML += `<li>${step}</li>`
+        return 2
+      })
+  }).then((step2) => {
+    return obtainInstruction('steak', step2)
+      .then((step) => {
+        document.querySelector("#steak").innerHTML += `<li>${step}</li>`
+        return 3
+      })
+  }).then((step3) => {
+    return obtainInstruction('steak', step3)
+      .then((step) => {
+        document.querySelector("#steak").innerHTML += `<li>${step}</li>`
+        return 4
+      })
+  }).then((step4) => {
+    return obtainInstruction('steak', step4)
+      .then((step) => {
+        document.querySelector("#steak").innerHTML += `<li>${step}</li>`
+        return 5
+      })
+  }).then((step5) => {
+    return obtainInstruction('steak', step5)
+      .then((step) => {
+        document.querySelector("#steak").innerHTML += `<li>${step}</li>`
+        return 6
+      })
+  }).then((step6) => {
+    return obtainInstruction('steak', step6)
+      .then((step) => {
+        document.querySelector("#steak").innerHTML += `<li>${step}</li>` 
+        document.querySelector("#steak").innerHTML += `<li>Stake is ready!</li>`
+        document.querySelector("#steakImg").removeAttribute('hidden') 
+      })
+  })
 // Iteration 3 using async/await
+async function makeBroccoli() {
+  await obtainInstruction('broccoli', 0)
+    .then((step) => {
+      document.querySelector("#broccoli").innerHTML += `<li>${step}</li>` 
+    })
+  await obtainInstruction('broccoli', 1)
+    .then((step) => {
+      document.querySelector("#broccoli").innerHTML += `<li>${step}</li>` 
+    })
+  await obtainInstruction (`broccoli`,2)
+    .then ((step)=> {
+    document.querySelector("#broccoli").innerHTML += `<li>${step}</li>`
+  })
+  await obtainInstruction (`broccoli`,3)
+    .then ((step)=> {
+    document.querySelector("#broccoli").innerHTML += `<li>${step}</li>`
+  })
+  await obtainInstruction('broccoli', 4)
+    .then((step) => {
+      document.querySelector("#broccoli").innerHTML += `<li>${step}</li>` 
+    })
+  await obtainInstruction (`broccoli`,5)
+    .then ((step)=> {
+    document.querySelector("#broccoli").innerHTML += `<li>${step}</li>`
+  })
+  await obtainInstruction (`broccoli`,6)
+    .then ((step)=> {
+    document.querySelector("#broccoli").innerHTML += `<li>${step}</li>`
+    document.querySelector("#broccoli").innerHTML += `<li> Broccoli is ready!</li>`
+    document.querySelector("#broccoliImg").removeAttribute(`hidden`)
+  })
+  
+};
+makeBroccoli()
+// Bonus 2 - Promise all
 // ...
+function getlist (steps){
```

Comment:
Here, you didn't need to create this getlist function.
Another way of doing that would be creating variables for each step of the obtainInstruction function, like:

```
const pr1 = obtainInstruction("brusselsSprouts", 0)
const pr2 = obtainInstruction("brusselsSprouts", 1)
const pr3 = obtainInstruction("brusselsSprouts", 2)
const pr4 = obtainInstruction("brusselsSprouts", 3)
const pr5 = obtainInstruction("brusselsSprouts", 4)
const pr6 = obtainInstruction("brusselsSprouts", 5)
const pr7 = obtainInstruction("brusselsSprouts", 6)
const pr8 = obtainInstruction("brusselsSprouts", 7)
```

and then on Promise.all(`[pr1, pr2, ...]`)